webapp/src/index.tsx

```diff
@@ -6,43 +6,75 @@ import type { GlobalState } from 'mattermost-redux/types/store'
 import type { Channel, ChannelMembership } from 'mattermost-redux/types/channels'
 import { Client4 } from 'mattermost-redux/client'
 
-import { ConferenceManager, type ConferenceConfig } from './services/conference-manager'
-import { App } from './App'
-import { MattermostManager } from './services/mattermost-manager'
-import { DisplayNameType, getPluginServerRoute, getPluginSettings, notifyJoinConference } from './utils'
 import { getConfig } from 'mattermost-redux/selectors/entities/general'
+import { DisplayNameType, type PluginSettings, getPluginSettings, notifyJoinConference, setMattermostStore, getMattermostStore } from './utils'
+import { App } from './App'
 
 import manifest from '../../plugin.json'
+import type { ConferenceConfig } from './types/ConferenceConfig'
+import { ConferenceContextProvider } from './contexts/ConferenceContext/ConferenceContext'
+
+import '@pexip/components/dist/style.css'
 
 const pluginId = manifest.id
 const icon = <i id='pexip-vmr-plugin-button' className='icon fa fa-video-camera'/>
 const dropDownText = 'Pexip Video Connect'
 
+let conferenceConfig: ConferenceConfig
+const RightHandSidebarComponent = (): JSX.Element => {
+  return (
+    <ConferenceContextProvider>
+      <App config={conferenceConfig} />
+    </ConferenceContextProvider>
+  )
+}
+
 class Plugin {
   private store: Store<GlobalState, Action>
   private rhsPlugin: RHSPlugin
 
   initialize (registry: PluginRegistry, store: Store<GlobalState, Action>): void {
     this.store = store
-    MattermostManager.setStore(store)
-    const script = document.createElement('script')
-    script.src = getPluginServerRoute(store.getState()) + '/public/pexrtc-32.js'
-    document.getElementsByTagName('head')[0].appendChild(script)
+    setMattermostStore(store)
     registry.registerChannelHeaderButtonAction(icon, this.action.bind(this), dropDownText)
-    this.rhsPlugin = registry.registerRightHandSidebarComponent(App, 'Pexip Video Connect')
+    this.rhsPlugin = registry.registerRightHandSidebarComponent(RightHandSidebarComponent, 'Pexip Video Connect')
```

Comment:
Maybe double-check this plugin name? At least in webapp3 we recently changed the naming. So I wonder if it should just say "Pexip"?

Reply:
Yes, we should change it at some point and also inform Mattermost of this change. They are using it in their Marketplace: https://mattermost.com/marketplace/pexip-video-connect/

Should I change it now? Only "Pexip" or any other name?